code/modules/recycling/disposal.dm

```diff
@@ -493,6 +493,38 @@
 	else
 		return ..(mover, target)
 
+
+
+/obj/machinery/disposal/wall
+	name = "inset disposal unit"
```

Comment:
did you test if this works? I am not sure if the bump proc is called when you are on a wall tile

Reply:
They work quite well, ive not noticed anything with bumping into them though. Is the bump proc tripping into one?